CHANGES.rst

```diff
@@ -16,6 +16,8 @@ New Features
 
 - ``astropy.convolution``
 
+  - Modified cython functions to release the GIL. [#3949]
```

Comment:
Actually, could you move this to the v1.0.4 changelog section instead?  There's no reason this can't be in that release.